odoo/models.py

```diff
@@ -1979,6 +1980,11 @@ def _read_group_fill_results(self, domain, groupby, remaining_groupbys,
         if not field.group_expand:
             return read_group_result
 
+        # the method defined in the group_expand parameter is either Model-specific or
+        # a Field-specific default
+        method_name = field.group_expand
+        group_expand = getattr(self, method_name, partial(getattr(field, method_name), self))
```

Comment:
This is a potential source of obscure bugs.
Adding or renaming a method on fields may accidentally screw up a field's `group_expand`; all you need is a name collision between *uncorrelated* classes (models vs fields).

I suggest to generalize the spec of `field.group_expand` pretty much like the compute method, i.e., it is either a method name (on the model) or a callable.  The adaptation would be
```python
group_expand = field.group_expand
if isinstance(group_expand, str):
    # group_expand is the name of a method on the model
    group_expand = getattr(type(self), group_expand)
assert callable(group_expand)
```
and then the call would pass `self` explicitly
```python
groups = group_expand(self, groups, domain, order)
```
while the selection field would default `group_expand` to `self._default_group_expand` (`self` being the field).